### PR TITLE
liquidator(Proxy-Helper-Scripts): Add DSProxy script to settle an expired position

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

While running the single reserve liquidator bot, we created a number of positions due to liquidations. As these contracts are now settling, we require a DSProxy helper script to let us settle these positions. This PR adds the smart contract and script to facilitate this settling of positions.

**Note**: I placed this script within the `trader` package so it would be alongside the other DSProxy helper scripts. This is a bad location and should be moved in future.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [X]  Untested

### CODE REVIEW

File: packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../common/implementation/FixedPoint.sol";
+
+// Simple contract used to Settle expired positions using a DSProxy.
+contract LiquidationWithdrawer {
```
Comment (nicholaspai) [inline packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol:7]: Why is this name `LiquidationWithdrawer` but the file is `PositionSettler`

File: packages/trader/scripts/SettleExpiredPositionWithDSProxy.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  // False as the second param in this function prevents the DSProxyManager from deploying a DSProxy if you don't have one.
+  await dsProxyManager.initializeDSProxy(argv.dsProxyAddress || null, false);
+
+  const dsProxyAddress = dsProxyManager.getDSProxyAddress();
+  if (!dsProxyAddress) throw new Error("DSProxy Address was not found or not parameterized");
+
+  // Create the callData as the encoded tx against the PositionSettler contract.
+  const PositionSettler = getTruffleContract("PositionSettler", web3);
+  const PositionSettlerInstance = new web3.eth.Contract(PositionSettler.abi);
+  const callData = PositionSettlerInstance.methods.redeem(argv.financialContractAddress).encodeABI();
```
Comment (nicholaspai) [inline packages/trader/scripts/SettleExpiredPositionWithDSProxy.ts:55]: This should be calling `settleExpired` not `redeem` right?

Comment (nicholaspai) [reply packages/trader/scripts/SettleExpiredPositionWithDSProxy.ts:55]: Also, should we not try to simulate the `settleExpired` and make sure it returns an `amountWithdrawn > 0`?

Comment (chrismaree) [reply packages/trader/scripts/SettleExpiredPositionWithDSProxy.ts:55]: yup. my bad

File: packages/trader/scripts/RedeemTokensWithDSProxy.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // This script enables a DSProxy to redeem tokens from a contract. This should be used after a swap/mint/liquidate transaction
 // to close out a bots position. To execute the script, run:
-// truffle exec./ scripts / RedeemTokensWithDSProxy.ts--financialContractAddress 0x123 --numTokens 100000000000000000000 --dsProxyFactoryAddress 0x123 --network mainnet_mnemonic
+// truffle exec ./scripts /RedeemTokensWithDSProxy.ts--financialContractAddress 0x123 --numTokens 100000000000000000000 --dsProxyFactoryAddress 0x123 --network mainnet_mnemonic
```
Comment (mrice32) [inline packages/trader/scripts/RedeemTokensWithDSProxy.ts:3]: weird spacing still...?

File: packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../common/implementation/FixedPoint.sol";
+
+// Simple contract used to Settle expired positions using a DSProxy.
+contract PositionSettler {
+    function settle(address financialContractAddress) public returns (FixedPoint.Unsigned memory) {
```
Comment (mrice32) [inline packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol:8]: Should we also put settleExpired in here? Or do we already have that one in a different script?

**Review by mrice32 — COMMENTED** [review]
Looks good!

File: packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol (hunk 4 above)
Comment (chrismaree) [reply packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol:8]: no you right. it should be settleExpired. I got confused for a second. this script is here to settle EMP positions minted by the liquidator.

File: packages/trader/scripts/RedeemTokensWithDSProxy.ts (hunk 3 above)
Comment (chrismaree) [reply packages/trader/scripts/RedeemTokensWithDSProxy.ts]: fixed.

File: packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol (hunk 4 above)
Comment (mrice32) [reply packages/core/contracts/proxy-scripts/bot-action-wrappers/PositionSettler.sol:8]: bump

**Review by mrice32 — APPROVED** [review]
LGTM once this last comment is fixed!

**Pull request merged**